src/shared/Core.Tests/SettingsTests.cs

```diff
@@ -280,6 +279,86 @@ public void Settings_IsTracingEnabled_EnvarPathy_ReturnsTrueOutValue()
             Assert.Equal(expectedValue, actualValue);
         }
 
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_EnvarUnset_ReturnsFalse()
+        {
+            var envars = new TestEnvironment();
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+            var result = settings.GetTrace2TracingEnabled(out string actualValue);
+
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_EnvarTruthy_ReturnsTrueOutValue()
+        {
+            const string expectedValue = "1";
+            var envars = new TestEnvironment
+            {
+                Variables = {[Constants.EnvironmentVariables.GitTrace2] = expectedValue}
+            };
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+            var result = settings.GetTrace2TracingEnabled(out string actualValue);
+
+            Assert.True(result);
+            Assert.Equal(expectedValue, actualValue);
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_EnvarFalsey_ReturnsFalseOutValue()
+        {
+            const string expectedValue = "0";
+            var envars = new TestEnvironment
+            {
+                Variables = {[Constants.EnvironmentVariables.GitTrace2] = expectedValue}
+            };
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+            var result = settings.GetTrace2TracingEnabled(out string actualValue);
+
+            Assert.False(result);
+            Assert.Equal(expectedValue, actualValue);
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_ConfigUnset_ReturnsFalse()
+        {
+            var envars = new TestEnvironment();
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+
+            Assert.False(settings.GetTrace2TracingEnabled(out string actualValue));
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_ConfigSet_ReturnsTrue()
+        {
+            const string section = Constants.GitConfiguration.Trace2.SectionName;
+            const string property = Constants.GitConfiguration.Trace2.EventTarget;
+
+            var envars = new TestEnvironment();
+            var git = new TestGit();
+
+            if (PlatformUtils.IsWindows())
+            {
+                git.Configuration.Global[$"{section}.{property}"] = new[] {"af_unix:./test.uds"};
+            }
+            else
+            {
+                git.Configuration.Global[$"{section}.{property}"] = new[] {"//./pipe/git-c780ac06-135a-4e9e-ab6c-d41e2d265baa"};
```

Comment:
Both "af_unix:./test.uds" and this "//./pipe/git-c78a...." string literals should not be in the source.  (They're not secrets or anything.)   However, they refer to whatever socket/pipe that we have the/a server listening on.

Is the goal here to actually use the pathnames and connect -- or just to confirm that they are non-empty (from an earlier test setup) (or return these values so that another test can confirm the correct value is returned) -- either way, I think we could use some fictional literals or something.

Reply:
The goal of the test is to show that if someone has the `trace2.eventtarget` config set, GCM will correctly detect that tracing is enabled. But totally fine with putting some dummy values in there.

Reply:
That's much better anyway because it removes the weird platform switch.